### DISCUSSION TITLE
Can I use other version of OMNET++ and INET?

### DISCUSSION THREAD

**Discussion by krzycho2** [eclipse/mosaic#97] [Q&A]
Hello. In _Network Simulator OMNeT++_ page (https://www.eclipse.org/mosaic/docs/simulators/network_simulator_omnetpp/) it's said that supported versions for OMNET++ and INET are 5.5 and 4.1. The current versions are 5.6.2 and 4.3.0 respectively. How can I use these versions in MOSAIC?

**Comment by kschrab** [comment]
MOSAIC is coupled with OMNeT++ using a federate, which is basically a proxy which exchanges protobuf-messages between the OmnetppAmbassador which is integrated in MOSAIC. This federate is written in C++ and extends the official OMNeT++ code with the code to send/read those protobuf-messages, but also loads the OMNeT++ simulator, adds and configures nodes, sends V2X messages, and so on. This federate code is located [in a separate repository](https://github.com/mosaic-addons/omnetpp-federate) and is only compatible with a specific version of OMNeT++ (currently 5.5). 

What you need to do would be to adjust the code in this repository to get it to work with the latest version of OMNeT++, and you also might need to adjust our installation script (located [here](https://github.com/eclipse/mosaic/tree/main/bundle/src/assembly/resources/fed/omnetpp)). You do not necessarily this script but you can build the federate by your own if you know what you're doing.
